plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Pom.java

```diff
@@ -89,23 +89,27 @@ public void addDependencies(List<Library> libraries) throws CoreException {
       dependencies = document.createElement("dependencies");
     }
     
-    //todo dedup
     for (Library library : libraries) {
       for (LibraryFile artifact : library.getLibraryFiles()) {
         Element dependency = document.createElement("dependency");
         MavenCoordinates coordinates = artifact.getMavenCoordinates();
         
-        Element groupId = document.createElement("groupId");
-        groupId.setTextContent(coordinates.getGroupId());
-        Element artifactId = document.createElement("artifactId");
-        artifactId.setTextContent(coordinates.getArtifactId());
-        Element version = document.createElement("version");
-        version.setTextContent(coordinates.getVersion());
-        dependency.appendChild(groupId);
-        dependency.appendChild(artifactId);
-        dependency.appendChild(version);
+        String groupId = coordinates.getGroupId();
+        String artifactId = coordinates.getArtifactId();
         
-        dependencies.appendChild(dependency);
+        if (!dependencyExists(dependencies, groupId, artifactId)) {
```

Comment:
A duplicate is not necessarily a mistake. E.g. this will catch cases where multiple libraries all add the same guava.jar.

Versions worth considering though I'm not sure what to do. Just pick the latest even though existing code may require the older version? We may need to leave this to the user to sort out. 

For scope, filed https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1486